src/popover/popover-default-animation.jsx

```diff
@@ -0,0 +1,140 @@
+import ThemeManager from '../styles/theme-manager';
+import Paper from '../paper';
+
+const PopoverAnimation = React.createClass({
+  mixins: [StylePropable],
+
+  propTypes: {
+    children: React.PropTypes.node,
+    className: React.PropTypes.string,
+    open: React.PropTypes.bool.isRequired,
+    style: React.PropTypes.object,
+    targetOrigin: PropTypes.origin,
+    zDepth: PropTypes.zDepth,
+  },
+
+  getInitialState() {
+    return {
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+      open: false,
+    };
+  },
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  getChildContext() {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+
+  getDefaultProps() {
+    return {
+      style: {},
+      zDepth: 1,
+    };
+  },
+
+  componentDidMount() {
+    this.setState({open:true}); //eslint-disable-line react/no-did-mount-set-state
+  },
+
+  componentWillReceiveProps(nextProps, nextContext) {
+    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+
+    this.setState({
+      open:nextProps.open,
+      muiTheme:newMuiTheme,
+    });
+  },
+
+
+  getStyles() {
+    let {targetOrigin} = this.props;
+    let horizontal = targetOrigin.horizontal.replace('middle', 'vertical');
+
+    return {
+      base: {
+        opacity:0,
+        transform:'scale(0, 0)',
+        transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+        position: 'fixed',
+        zIndex: this.state.muiTheme.zIndex.popover,
+        transition: Transitions.easeOut('250ms', ['transform', 'opacity']),
+        maxHeight:'100%',
+
+      },
+      horizontal: {
+        maxHeight:'100%',
+        overflowY:'auto',
+        transform:'scaleX(0)',
+        opacity:0,
+        transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+        transition: Transitions.easeOut('250ms', ['transform', 'opacity']),
+      },
+      vertical: {
+        opacity:0,
+        transform:'scaleY(0)',
+        transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
+        transition: Transitions.easeOut('500ms', ['transform', 'opacity']),
+      },
+    };
+  },
+
+  getOpenStyles() {
+    return {
+      base: {
+        opacity: 1,
+        transform:'scale(1, 1)',
+      },
+      horizontal: {
+        opacity: 1,
+        transform:'scaleX(1)',
+      },
+      vertical: {
+        opacity: 1,
+        transform:'scaleY(1)',
+      },
+    };
+  },
+
+  render() {
+    let {
+      className,
+      style,
+      zDepth,
+    } = this.props;
+
+    let styles = this.getStyles();
+    let openStyles = {};
+    if (this.state.open)
+      openStyles = this.getOpenStyles();
+
+    return (
+      <Paper
+        style={this.prepareStyles(styles.base, style, openStyles.base)}
```

Comment:
First warning solved by using `this.mergeStyles` here.
(By the way, this method shouldn't be in the mixin since it's an utility function cc @subjectix)